libs/core/kiln_ai/adapters/extractors/gemini_extractor.py

```diff
@@ -0,0 +1,152 @@
+        "application/pdf",
+        "application/x-javascript",
+        "text/javascript",
+        "application/x-python",
+        "text/x-python",
+        "text/plain",
+        "text/markdown",  # not officially listed, but works
+        "text/html",
+        "text/css",
+        "text/md",
+        "text/csv",
+        "text/xml",
+        "text/rtf",
+    ],
+    Kind.IMAGE: [
+        "image/png",
+        "image/jpeg",
+        "image/webp",
+        "image/heic",
+        "image/heif",
+    ],
+    Kind.VIDEO: [
+        "video/mp4",
+        "video/mpeg",
+        "video/mov",
+        "video/avi",
+        "video/x-flv",
+        "video/mpg",
+        "video/webm",
+        "video/wmv",
+        "video/3gpp",
+    ],
+    Kind.AUDIO: [
+        "audio/wav",
+        "audio/mp3",
+        "audio/mpeg",  # not explicitly supported, but mimetypes stdlib returns audio/mpeg for MP3
+        "audio/aiff",
+        "audio/aac",
+        "audio/ogg",
+        "audio/flac",
+    ],
+}
+
+
+class GeminiExtractorConfig(BaseExtractorConfig):
+    default_prompt: str = Field(
+        description="The default prompt to use for the extractor."
+    )
+    custom_prompt: str | None = Field(
+        default=None,
+        description="A custom prompt provided by the user to use for the extractor. It takes precedence over the prompt_for_kind and default_prompt.",
+    )
+    prompt_for_kind: dict[Kind, str] = Field(
+        default_factory=dict,
+        description="A dictionary of prompts for each kind of file.",
+    )
+    model: str = Field(description="The model to use for the extractor.")
+    output_format: ExtractionFormat = Field(
+        default=ExtractionFormat.MARKDOWN,
+        description="The format to use for the output.",
+    )
+
+
+class GeminiExtractor(BaseExtractor):
+    def __init__(self, gemini_client: genai.Client, config: GeminiExtractorConfig):
+        super().__init__(config)
+        self.gemini_client = gemini_client
+
+        # hack to access the concrete config subclass here without failing type checking
+        self.config = config
+
+    def _get_kind_from_mime_type(self, mime_type: str) -> Kind:
+        """
+        Determines the file kind based on its MIME type.
+
+        Returns:
+            The corresponding Kind enum value for the given MIME type.
+
+        Raises:
+            ValueError: If the MIME type is not supported.
+        """
+        for kind, mime_types in MIME_TYPES_SUPPORTED.items():
+            if mime_type in mime_types:
+                return kind
+        raise ValueError(f"Unsupported MIME type: {mime_type}")
+
+    def _get_prompt_for_kind(self, kind: Kind) -> str:
+        """
+        Returns the prompt string for the specified file kind.
+
+        If a prompt is configured for the given kind, it is returned; otherwise, the extractor's default prompt is used.
+        """
+        return self.config.prompt_for_kind.get(kind, self.config.default_prompt)
```

Comment:
per other comment, the extractor shouldn't be considering "default". Keep that in the Config init or UI.